### ISSUE TITLE
Expose code change webhook from `sleuth_code_change_source` resource

### ISSUE THREAD

**Issue by davejbax** [sleuth-io/terraform-provider-sleuth#200]
We're managing our Sleuth code change sources using this provider, which works great for the most part!

We'd also like to manage the GitLab project webhooks from within the same set of Terraform. The GitLab Terraform provider allows us to create these, but the Sleuth provider does not currently expose the webhook URL or secret for a given code change source.

As I understand, the webhook URL/secret _is_ exposed by Sleuth's GraphQL API. Would it be possible to add these as attributes exposed by a `sleuth_code_change_source`?

**Comment by zigomir** [comment]
Sleuth's GraphQL does expose a `repository` field via `CodeChangeSource > RepositoryType` which includes `webhook` field. That webhook is only meant for creating a webhook that Sleuth itself uses for tracking repository events (i.e. PR merged) and not as a generic webhook. That's why we don't expose webook related fields in a `createCodeChangeSource` mutation's input.

Why would you prefer using Sleuth's TF provider over GitLab one for managing GitLab project webhooks?

**Comment by davejbax** [comment]
> That webhook is only meant for creating a webhook that Sleuth itself uses for tracking repository events (i.e. PR merged)

Sorry, I should have clarified: we're using GitLab integration in Sleuth in [read-only mode](https://help.sleuth.io/integrations-1/code-deployment/gitlab). Hence, if I'm not misunderstanding, we need to set up the same webhook ourselves.

> in a createCodeChangeSource mutation's input.

To clarify, the issue is not about providing a custom webhook URL to the Sleuth provider, but exposing the webhook URL as an _output_ attribute of the code change source resource. The intention is that our Terraform would create the code change resource, get the webhook URL Sleuth has created along with it, and then set this URL in the GitLab repository ourselves using the GitLab provider (since we are using read-only mode).

Does that help to explain the issue, or am I misunderstanding your question?

**Comment by zigomir** [comment]
np, sorry for misunderstanding from my side.

(I'm not too familiar with TF and am sorry if am asking silly questions)

So defining the `sleuth_code_change_source` block you'd like to see if it would've returned `webhook` URL and webhook secret (we call it API key on our side) so you could use it for other GitLab specific TF definitions?

i.e.:

```tf
# creates code change source in Sleuth
resource "sleuth_code_change_source" "cd" {
  project_slug = sleuth_project.tf.slug
  name = "foo"
  repository = {
     …
  }
}

# Does something in GitLab…
resource "something_gitlab" "gl" {
  webhook = sleuth_code_change_source.cd.webhook
  webhook_secret = sleuth_code_change_source.cd.webhook_secret
}
```

**Comment by davejbax** [comment]
Yes, exactly that! 😄  The GitLab resource we'd be using for reference is [`gitlab_project_hook`](https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs/resources/project_hook)

**Comment by zigomir** [comment]
Got it 👍  I'll see what we can do about it and will keep you updated.
